array.go

```diff
@@ -71,7 +71,7 @@ func NewArray(tdbCtx *Context, uri string) (*Array, error) {
 	if ret != C.TILEDB_OK {
 		return nil, fmt.Errorf("error creating tiledb array: %w", array.context.LastError())
 	}
-	freeOnGC(&array)
+	runtime.AddCleanup(&array, freeFreeable, Freeable(&array))
```

Comment:
I've been thinking of another approach that will avoid the `runtime.AddCleanup` boilerplate everywhere and maybe support both old and new-style finalizers.

---

Also we should keep the `Cleanup` object somewhere so that we remove it if the object is manually `Free()`ed.

Reply:
Can you create a story? I can revert for now, and we can sync on this tomorrow morning or you can pick it up if you'd like :slightly_smiling_face:

Reply:
Opened SC-64250.